### DISCUSSION TITLE
MQTT events not working with 0.14

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [blakeblackshear/frigate#12227] [General]
Since rebuilding to 0.14, I can't get MQTT events working.

There are 40 of these frigate log messages:
`
frigate014   | 2024-06-30 13:46:56.858004632  2024/06/30 13:46:56 [error] 454#454: *1 connect() failed (111: Connection refused) while connecting to upstream, client: ::ffff:172.19.210.68, server: , request: "GET /api/stats HTTP/1.1", subrequest: "/auth", upstream: "http://127.0.0.1:5001/auth", host: "172.19.210.11:5000"
`
and then this:
`frigate014   | 2024-06-30 16:48:59.206537642  [2024-06-30 13:48:02] frigate.comms.mqtt             DEBUG   : MQTT connected
`

Then MQTT log shows a successful connection (and never a disconnect):
`2024-06-30 13:48:02: New connection from 172.19.210.11:38375 on port 1883.
2024-06-30 13:48:02: New client connected from 172.19.210.11:38375 as frigate (p2, c1, k60, u'frigate_mqtt').`

and the HA core log shows
`
2024-06-30 13:48:01.859 DEBUG (MainThread) [custom_components.frigate] Finished fetching frigate data in 0.006 seconds (success: False)
2024-06-30 13:48:06.907 INFO (MainThread) [custom_components.frigate] Fetching frigate data recovered
2024-06-30 13:48:06.907 DEBUG (MainThread) [custom_components.frigate] Finished fetching frigate data in 0.054 seconds (success: True)`

But no frigate listener or events are ever shown (I check through HA | Developer Tools - I dont see a frigate listener and listening for "frigate" events shows nothing).

Environment:
- System 1 (172.19.210.68): HA 2024.6.4 with Mosquitto 6.4.1 addon plus local HA user frigate_mqtt and Frigate integration 5.2.0
- System 2 (172.19.210.11): Docker Frigate 0.14.0-4e5a6eb

Observations:
* MQTT addon is running and shows listeners and events from other clients
* Frigate and frigate HA integration is running (camera feeds, alerts, recordings, clips, etc)
* Restarting Frigate, HA, MQTT, reinstalling Frigate integration or MQTT, adding/removing logins to the MQTT config makes no difference

Frigate Integration URL:
http://172.19.210.11:5000/

HA MQTT config:
`require_certificate: false
certfile: fullchain.pem
keyfile: privkey.pem
customize:
  active: false
  folder: mosquitto
`

Frigate MQTT config:
`mqtt:
  host: 172.19.210.68
  password: something
  user: frigate_mqtt`

**Comment by NickM-27** [comment]
Best to use MQTT explorer

**Reply by JesseWebDotCom** [reply]
You are right - I see frigate messages coming in through MQTT explorer.  Odd how they are not visible in HA Developer Tools (I actually see them in the HA addon Node-Red).

**Reply by JesseWebDotCom** [reply]
https://mqtt-explorer.com/
